starlite/signature/models.py

```diff
@@ -76,12 +76,18 @@ def is_mapping(self) -> bool:
     @property
     def is_non_string_iterable(self) -> bool:
         """Check if the field type is an Iterable."""
-        return is_non_string_iterable(self.field_type)
+        field_type = self.field_type
+        if self.is_optional:
+            field_type = make_non_optional_union(field_type)
+        return is_non_string_iterable(field_type)
```

Comment:
Would it not be better to have this in `utils.predicates.is_non_string_iterable`?

Reply:
You mean move the optional stripping logic into `is_non_string_iterable()`?

Reply:
Yes. Seems to make the most sense to me, since they'd also benefit from this additional check.

Reply:
I suppose it boils down to whether (for example) `list[Any] | None` should be considered as a non-string sequence/iterable in the general case, or not.

Imagine a new contributor comes across `utils.predicates.is_non_string_iterable()` while doing some unrelated work, and they have no context of this issue, or these changes. Could it not be surprising that it also de-optionalizes the provided type, before testing whether it is an iterable?

The same argument could also be made about `SignatureField.is_non_string_iterable`, but at least the scope of use is somewhat narrowed to signature modelling here.

Reply:
> I suppose it boils down to whether (for example) list[Any] | None should be considered as a non-string sequence/iterable in the general case, or not.

It shouldn't, which means it's fine if this stays where it is. 

> The same argument could also be made about SignatureField.is_non_string_iterable, but at least the scope of use is somewhat narrowed to signature modelling here.

I was going to bring this up, which makes me think, maybe this could be solved by just giving it a different name?
Since effectively it does *not* just tell you if it's a non-string iterable. It's more like `is_non_string_iterable_or_non_string_iterable_none_union`, which admittedly is a mouthful, but I think *something* to indicate that this will unwrap `None` unions may be good.

Reply:
well I made note of the behavior in the docstring, and referenced the issue as to why we do it.. It looks like we don't document these so if a good name for the property comes to me in my sleep, it should be safe enough to change.